Approx::operator() not const-correct

**Describe the bug**

The `Approx` type has an overload of `template <typename T, ...SFINAE...> Approx operator()(T const&)` which (correct me if I'm wrong) is meant to be a factory function for instances that have the same epsilon, margin, and scale, but that use the passed value. 

AFAICT this should be const on the instance, but it's not.

Minimum failing example:
```C++
#include <catch2/catch.hpp>

TEST_CASE("Approx factory is const-correct") {
  // Set up a template Approx with problem-specific margin, etc.
  Approx const apprx = Approx(0.0).margin(1e-6);
  double value = 1.0;
  // Use template in assertions
  REQUIRE(value == apprx(1.0));
}
```

**Expected behavior**
Above test compiles, runs and passes.

**Reproduction steps**
See above.

**Platform information:**
 - OS: RHEL 8
 - Compiler+version: GCC 8.2.0
 - Catch version: 2.13.6


**Additional context**
Add any other context about the problem here.

Sorry forgot to add error message:
```
$ g++ -I$Catch2_DIR/include const-approx.cpp 
In file included from const-approx.cpp:1:
const-approx.cpp: In function ‘void ____C_A_T_C_H____T_E_S_T____0()’:
const-approx.cpp:6:29: error: no match for call to ‘(const Catch::Detail::Approx) (double)’
   REQUIRE(value == apprx(1.0));
                             ^
In file included from const-approx.cpp:1:
$Catch2_DIR/include/catch2/catch.hpp:3094:16: note: candidate: ‘Catch::Detail::Approx Catch::Detail::Approx::operator()(const T&) [with T = double; <template-parameter-1-2> = void]’ <near match>
         Approx operator()( T const& value ) {
                ^~~~~~~~
$Catch2_DIR/include/catch2/catch.hpp:3094:16: note:   passing ‘const Catch::Detail::Approx*’ as ‘this’ argument discards qualifiers
```

--- a/include/internal/catch_approx.h
+++ b/include/internal/catch_approx.h
@@ -33,7 +33,7 @@ namespace Detail {
         Approx operator-() const;
 
         template <typename T, typename = typename std::enable_if<std::is_constructible<double, T>::value>::type>
-        Approx operator()( T const& value ) {
+        Approx operator()( T const& value ) const {
             Approx approx( static_cast<double>(value) );
             approx.m_epsilon = m_epsilon;
             approx.m_margin = m_margin;

--- a/projects/SelfTest/UsageTests/Approx.tests.cpp
+++ b/projects/SelfTest/UsageTests/Approx.tests.cpp
@@ -212,4 +212,11 @@ TEST_CASE( "Comparison with explicitly convertible types", "[Approx]" )
 
 }
 
+TEST_CASE("Approx::operator() is const correct", "[Approx][.approvals]") {
+  const Approx ap = Approx(0.0).margin(0.01);
+
+  // As long as this compiles, the test should be considered passing
+  REQUIRE(1.0 == ap(1.0));
+}
+
 }} // namespace ApproxTests

